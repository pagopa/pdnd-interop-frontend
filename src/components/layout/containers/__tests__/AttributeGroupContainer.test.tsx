import React from 'react'
import { render } from '@testing-library/react'
import { AttributeGroupContainer } from '../AttributeGroupContainer'

describe('AttributeGroupContainer', () => {
  it('should match snapshot', () => {
    const { container } = render(<AttributeGroupContainer groupNum={1}>{}</AttributeGroupContainer>)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot with header content', () => {
    const { container } = render(
      <AttributeGroupContainer groupNum={1} headerContent={<></>}>
        {}
      </AttributeGroupContainer>
    )
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot with footer content', () => {
    const { container } = render(
      <AttributeGroupContainer groupNum={1} footerContent={<></>}>
        {}
      </AttributeGroupContainer>
    )
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot with both footer and header content', () => {
    const { container } = render(
      <AttributeGroupContainer groupNum={1} headerContent={<></>} footerContent={<></>}>
        {}
      </AttributeGroupContainer>
    )
    expect(container).toMatchSnapshot()
  })
})
