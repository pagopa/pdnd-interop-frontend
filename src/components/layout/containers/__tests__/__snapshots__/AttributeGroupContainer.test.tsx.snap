// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AttributeGroupContainer > should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-ydo5m"
  >
    <div
      class="css-42v4ct-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
      >
        title
      </p>
      <div
        class="MuiBox-root css-0"
      />
    </div>
    <div
      class="MuiBox-root css-hhy13k"
    />
  </div>
</div>
`;

exports[`AttributeGroupContainer > should match snapshot with both footer and header content 1`] = `
<div>
  <div
    class="MuiBox-root css-ydo5m"
  >
    <div
      class="css-42v4ct-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
      >
        title
      </p>
      <div
        class="MuiBox-root css-0"
      />
    </div>
    <div
      class="MuiBox-root css-hhy13k"
    />
    <div
      class="MuiBox-root css-1wr9hal"
    />
  </div>
</div>
`;

exports[`AttributeGroupContainer > should match snapshot with footer content 1`] = `
<div>
  <div
    class="MuiBox-root css-ydo5m"
  >
    <div
      class="css-42v4ct-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
      >
        title
      </p>
      <div
        class="MuiBox-root css-0"
      />
    </div>
    <div
      class="MuiBox-root css-hhy13k"
    />
    <div
      class="MuiBox-root css-1wr9hal"
    />
  </div>
</div>
`;

exports[`AttributeGroupContainer > should match snapshot with header content 1`] = `
<div>
  <div
    class="MuiBox-root css-ydo5m"
  >
    <div
      class="css-42v4ct-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
      >
        title
      </p>
      <div
        class="MuiBox-root css-0"
      />
    </div>
    <div
      class="MuiBox-root css-hhy13k"
    />
  </div>
</div>
`;
